src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndex.cs

```diff
@@ -77,7 +77,8 @@ public override IIndexedDocumentsEnumerator GetMapEnumerator(IEnumerable<Documen
 
         public override void Update(IndexDefinitionBase definition, IndexingConfiguration configuration)
         {
-            throw new NotSupportedException($"{Type} index does not support updating it's definition and configuration.");
+            SetLock(definition.LockMode);
+            SetPriority(definition.Priority);
```

Comment:
Something that was inherited from the base code of index, but for auto-indexes it does not make sense anymore I think.